pkg/ast/spl/spl.peg

```diff
@@ -3295,10 +3433,102 @@ EarliestOnly <- CMD_EARLIEST EQUAL earliestTime:AbsoluteTimestamp !. {
     return node, nil
 }
 
-AbsoluteTimestamp <- [0-9][0-9] '/' [0-9][0-9] '/' [0-9][0-9][0-9][0-9] ':' [0-9][0-9] ':' [0-9][0-9] ':' [0-9][0-9] {
+RelIntegerAsString <- [-+][0-9]* {
+    return string(c.text), nil
+}
+
+WeekSnap <- ('w'[0-7]) {
     return string(c.text), nil
 }
 
+RelTimeUnit <- timeUnit:(Second / Month / Minute / Hour / Day / Week / Quarter / Year) {
+    return strconv.Itoa(int(timeUnit.(utils.TimeUnit))), nil
+}
+
+Snap <- '@' snap:(WeekSnap / RelTimeUnit) {
+    return snap.(string), nil
+}
+
+Offset <- off:(RelIntegerAsString) tuOff:(RelTimeUnit) {
+    var offsetNum int64 = 0
+    var err error
+    offStr := off.(string)
+    if offStr == "+" {
+        offsetNum = 1
+    } else if offStr == "-" {
+        offsetNum = -1
+    } else {
+        offsetNum, err = strconv.ParseInt(offStr, 10, 64)
+        if err != nil {
+            return nil, fmt.Errorf("Invalid Relative Time Format time unit value: (%v): , err: %v", offStr, err)
+        }
+    }
+
+    timeUnit, err := strconv.Atoi(tuOff.(string))
+    if err != nil {
+        return nil, fmt.Errorf("Invalid Time Unit value: (%v), err: %v", tuOff.(string), err)
+    }
+
+    relTimeOffset := ast.RelativeTimeOffset{
+        Offset: offsetNum,
+        TimeUnit: utils.TimeUnit(timeUnit),
+    }
+
+    return relTimeOffset, nil
+}
+
+
+
+// TODO: Chained relative modifiers
+RelativeTimestamp <- defaultTime:("now" / "1")? offset:(Offset)? snapParam:(Snap)? {
```

Comment:
Why are `"now"` and `"1"` the only valid `defaultTime` values, but not other digits? Do you have a splunk documentation link? It could be helpful to add that as a comment here

Reply:
you can pass now to set to the currenTime and 1 is the unix epoch 1. These are the only constants that are there on [doc](https://docs.splunk.com/Documentation/Splunk/9.1.2/Search/Specifytimemodifiersinyoursearch)

Reply:
Ok. Can you add that link as a comment in the peg file?

Reply:
addressed in the latest commit